### PR TITLE
Add multi_cn

### LINKED ISSUE #3533: multi corpus Chinese ASR recipe
There are over 1k hours of free Chinese data on openslr. A new recipe using all the available Chinese free corpus will demonstrate how to:
1. merge and train Chinese models using data from multiple sources
2. merge vocabulary from multiple sources
3. train a general LM and adapt to certain domain.

### PR DESCRIPTION
**Pull request by naxingyu**
This is a WIP recipe, after merging, it closes #3533 .
For the idea and discussion, refer to #3533

### CODE REVIEW

**Comment by naxingyu** [comment]
GMM stage results are archived. I'm waiting for chain model results.
@greenbu please verify the results on AISHELL server.

**Comment by greenbu** [comment]
OK, I will do and feedback.
________________________________

**Comment by greenbu** [comment]
We verified the results on AISHELL server (four GPUs [TITAN Xp], memory 256G), and the all results are list as below:
#GMM results
# aishell test set results
%WER 22.35 [ 23414 / 104765, 683 ins, 1502 del, 21229 sub ] exp/tri1b/decode_aishell_test_tg/cer_13_0.0
%WER 20.66 [ 21641 / 104765, 715 ins, 1398 del, 19528 sub ] exp/tri2a/decode_aishell_test_tg/cer_13_0.0
%WER 18.88 [ 19781 / 104765, 748 ins, 1123 del, 17910 sub ] exp/tri3a/decode_aishell_test_tg/cer_13_0.0
%WER 17.33 [ 18157 / 104765, 762 ins, 997 del, 16398 sub ] exp/tri4a/decode_aishell_test_tg/cer_13_0.5
%WER 14.59 [ 15280 / 104765, 721 ins, 643 del, 13916 sub ] exp/tri4a_cleaned/decode_aishell_tg/cer_14_0.0

# aidatatang test set results
%WER 34.91 [ 163717 / 468933, 4383 ins, 33652 del, 125682 sub ] exp/tri1b/decode_aidatatang_test_tg/cer_14_0.0
%WER 33.99 [ 159374 / 468933, 4627 ins, 31327 del, 123420 sub ] exp/tri2a/decode_aidatatang_test_tg/cer_14_0.0
%WER 23.61 [ 110733 / 468933, 4558 ins, 19241 del, 86934 sub ] exp/tri3a/decode_aidatatang_test_tg/cer_14_0.0
%WER 20.20 [ 94730 / 468933, 4518 ins, 16400 del, 73812 sub ] exp/tri4a/decode_aidatatang_test_tg/cer_14_0.0
%WER 17.87 [ 83797 / 468933, 5453 ins, 14562 del, 63782 sub ] exp/tri4a_cleaned/decode_aidatatang_tg/cer_16_0.0

# magicdata test set results
%WER 28.87 [ 69278 / 239927, 6050 ins, 14123 del, 49105 sub ] exp/tri1b/decode_magicdata_test_tg/cer_17_0.0
%WER 28.69 [ 68847 / 239927, 6111 ins, 12611 del, 50125 sub ] exp/tri2a/decode_magicdata_test_tg/cer_17_0.0
%WER 22.31 [ 53536 / 239927, 5993 ins, 8155 del, 39388 sub ] exp/tri3a/decode_magicdata_test_tg/cer_17_0.5
%WER 15.60 [ 37424 / 239927, 3676 ins, 5398 del, 28350 sub ] exp/tri4a/decode_magicdata_test_tg/cer_17_0.5
%WER 13.88 [ 33292 / 239927, 5727 ins, 3504 del, 24061 sub ] exp/tri4a_cleaned/decode_magicdata_tg/cer_17_0.0

# thchs test set results
%WER 35.84 [ 29083 / 81139, 352 ins, 1851 del, 26880 sub ] exp/tri1b/decode_thchs_test_tg/cer_11_0.5
%WER 32.86 [ 26662 / 81139, 331 ins, 1684 del, 24647 sub ] exp/tri2a/decode_thchs_test_tg/cer_11_0.5
%WER 30.41 [ 24678 / 81139, 315 ins, 1434 del, 22929 sub ] exp/tri3a/decode_thchs_test_tg/cer_11_0.5
%WER 27.86 [ 22609 / 81139, 415 ins, 1269 del, 20925 sub ] exp/tri4a/decode_thchs_test_tg/cer_10_0.5
%WER 25.37 [ 20584 / 81139, 397 ins, 831 del, 19356 sub ] exp/tri4a_cleaned/decode_thchs_tg/cer_10_1.0

#Chain results
# aishlell test set result
%WER 5.93 [ 6216 / 104765, 120 ins, 154 del, 5942 sub ] exp/chain/tdnn_cnn_1a_sp/decode_aishell_tg/cer_10_1.0

# aidatatang test set result
%WER 4.96 [ 23279 / 468933, 1867 ins, 3206 del, 18206 sub ] exp/chain/tdnn_cnn_1a_sp/decode_aidatatang_tg/cer_10_0.0

# magicdata test set result
%WER 3.94 [ 9443 / 239927, 1048 ins, 1779 del, 6616 sub ] exp/chain/tdnn_cnn_1a_sp/decode_magicdata_tg/cer_11_0.5

# thchs test set results
%WER 12.80 [ 10387 / 81139, 135 ins, 278 del, 9974 sub ] exp/chain/tdnn_cnn_1a_sp/decode_thchs_tg/cer_9_1.0

# Chain_online result
# aishlell test set result
%WER 5.94 [ 6228 / 104765, 124 ins, 158 del, 5946 sub ] exp/chain/tdnn_cnn_1a_sp_online/decode_aishell_tg/cer_11_1.0

# aidatatang test set result
%WER 4.96 [ 23263 / 468933, 1877 ins, 3200 del, 18186 sub ] exp/chain/tdnn_cnn_1a_sp_online/decode_aidatatang_tg/cer_10_0.0

# magicdata test set result
%WER 3.95 [ 9474 / 239927, 1070 ins, 1769 del, 6635 sub ] exp/chain/tdnn_cnn_1a_sp_online/decode_magicdata_tg/cer_11_0.5

# thchs test set results
%WER 12.77 [ 10362 / 81139, 135 ins, 277 del, 9950 sub ] exp/chain/tdnn_cnn_1a_sp_online/decode_thchs_tg/cer_9_1.0

**Comment by naxingyu** [comment]
@danpovey This recipe is ready for review. The results have been verified independently.

**Comment by danpovey** [comment]
Thanks!  I don't see any enormous files or anything that can't be fixed after check-in, so I'll assume it's been debugged and merge it right away.

**Comment by danpovey** [comment]
BTW, something that was mentioned to me in my recent trip to China is that people are having good luck with getting rid of pitch, letting the nnet learn it implicitly from the features.  (Not sure if these were higher-resolution mel bins, e.g. 80 bins.)  Something that was mentioned to me, was that to get it to work for children's speech, you have to mess with the pitch using sox, as an augmentation method, to ensure that the training data contains high pitches.

**Comment by naxingyu** [comment]
Thanks! Getting rid of pitch is definitely worth trying, especially for online serving. Actually I am getting some positive results by getting rid of pitch in multi_cn. I'll keep investigate and update if results are promising.

**Comment by chen1399** [comment]
> This is a WIP recipe, after merging, it closes #3533 .
> For the idea and discussion, refer to #3533

Hi, audior. How can I try using my own voice?
I used 
``` online-wav-gmm-decode-faster --verbose=1 --rt-min=0.8 --rt-max=0.85\
            --max-active=4000 --beam=12.0 --acoustic-scale=0.0769 --left-context=3 --right-context=3 \
            scp:$decode_dir/input.scp $ac_model/final.mdl $ac_model/HCLG.fst \
            $ac_model/words.txt '1:2:3:4:5' ark,t:$decode_dir/trans.txt \
            ark,t:$decode_dir/ali.txt $trans_matrix;;
```
there is a error:
``` ERROR (online-wav-gmm-decode-faster[5.5]:ExpectToken():io-funcs.cc:209) Expected token "<DIMENSION>", got instead "<Nnet3>". ```

**Comment by chen1399** [comment]
> This is a WIP recipe, after merging, it closes #3533 .
> For the idea and discussion, refer to #3533

Hi, author. How can I try using my own voice?
I used
```
online-wav-gmm-decode-faster --verbose=1 --rt-min=0.8 --rt-max=0.85\
            --max-active=4000 --beam=12.0 --acoustic-scale=0.0769 --left-context=3 --right-context=3 \
            scp:$decode_dir/input.scp $ac_model/final.mdl $ac_model/HCLG.fst \
            $ac_model/words.txt '1:2:3:4:5' ark,t:$decode_dir/trans.txt \
            ark,t:$decode_dir/ali.txt $trans_matrix;;
```
there is a error:
``` ERROR (online-wav-gmm-decode-faster[5.5]:ExpectToken():io-funcs.cc:209) Expected token "<DIMENSION>", got instead "<Nnet3>".```
